### DISCUSSION TITLE
[教程]如何配置增强导出

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#2958] [同步助手]
# 重要提示

> ### 嗨，简悦全新的 [教程中心](https://kb.simpread.pro/) 已上线，**下面的内容已经已同步更新**，建议 [访问这里](https://kb.simpread.pro/#/page/增强导出) 查看。

***

> 增强导出是简悦同步助手的核心功能，通过此功能可以衍生出 👉  [资料库](https://github.com/Kenshin/simpread/discussions/2956)，即：通过简悦，可以让你在**坚果云或 Dropbox 特定文件夹的内容在线查看**。

## 使用前提

1. 如果你是简悦的重度使用者，一般来说都需要配合自动同步功能 https://github.com/Kenshin/simpread/discussions/2754

2. 增强导出可以设置任意文件夹，但如果你想要使用 https://github.com/Kenshin/simpread/discussions/2956 功能的话，则需要按照 `<网盘根目录>/SimpRead/output` 目录结构，不可更改。

3. 下文均以使用资料库为前提，如不是使用资料库的话，可更改为任意文件夹。

## 配置增强导出

> 在同步助手中指定 `output` 位置

![image](https://user-images.githubusercontent.com/81074/138262208-1e3c4ce6-ae48-42b0-996a-1afc057a4bab.png)

> **简悦 · 同步助手** 开启相关选项

![image](https://user-images.githubusercontent.com/81074/138263177-2d0743c7-9a90-475b-bf18-aa396dbcd2b4.png)

> **简悦（扩展端） → 选项页 → 共通 → 简悦 · 同步助手** 也同样开启以下功能

![image](https://user-images.githubusercontent.com/81074/138262893-11f18619-e4d7-4716-a84b-f73ca709ea12.png)

> 同时确保 `output` 与同步盘保持下图关系。

![image](https://user-images.githubusercontent.com/81074/138262363-cb835447-4c9d-47dd-a98d-37d146e5af6d.png)

## 引申

> 通过配置 [自动化](http://ksria.com/simpread/docs/#/自动化) 功能，可方便的在简悦的阅读模式中将任意内容导入到 `output` 文件夹中。

## 关联

- [如何让稍后读支持读取本地缓存 #2146](https://github.com/Kenshin/simpread/discussions/2146)
- [当产生标注时，如何自动同步到本地 #2220](https://github.com/Kenshin/simpread/discussions/2220)
- [当产生标注时，如何同时下载标注与原文 #2147](https://github.com/Kenshin/simpread/discussions/2147)
- [让简悦变成你的知识库（离线且具有永久链接） #2221](https://github.com/Kenshin/simpread/discussions/2221)
- [将自动化服务变成你的导出服务 #2151](https://github.com/Kenshin/simpread/discussions/2151)

**Comment by Kenshin** [comment]
# 增强导出

```
{"extension":"external", "path":"/Users/xxxx/xxxx/simpublish-demo/api/_output"}
{"extension":"pdf", "path":"/Users/xxxx/xxxx/Ebook"}
{"extension":"epub", "path":""}
{"extension":"docx", "path":""}
{"extension":"assets", "path":"/Users/xxxx/xxxx/Obsidian/SimpRead"}
{"extension":"textbundle", "path":""}
{"extension":"md", "path":"/Users/xxxx/xxxx/Obsidian/SimpRead"}
```

当导出类型为 Markdown 且为标注时，保存到 `/Users/xxxx/xxxx/Obsidian@simpread/SimpRead`

```
{"extension":"md", "path":"/Users/xxxx/xxxx/Obsidian@simpread/SimpRead", "type":"annote"}
```

当导出类型为 md + assets 且为全文时，保存到 `/Users/xxxx/xxxx/Obsidian@simpread/Inbox`

```
{"extension":"assets", "path":"/Users/xxxx/xxxx/Obsidian@simpread/Inbox", "type":"page"}
```

当导出类型为 `tts`  ，保存到 `/Users/xxxx/xxxx/tts`

```
{"extension":"tts", "path":"/Users/xxxx/xxxx/tts"}
```

简单的解释

- `extension` → 任意文件的后缀 `epub` `docx` `md` `html` 等

  有几个特殊的后缀：

  - `external` → 导出 HTML 时，将当前 `xxx.html` 复制到 `path` 对应的目录，此功能注意针对 [简悦的外部资料库的自建方案](https://github.com/Kenshin/simpread/discussions?discussions_q=label%3A%22self+built%22)。

  - `assets` → 可以辅助 [导出 md + assets 格式](https://github.com/Kenshin/simpread/discussions/4041) 实现 `md + assets` 结构。

  - `tts` → 可以辅助 [朗读](TTS?id=下载位置) 功能将正文转换的语音保存到任意位置。

- `output` → 导出文件夹地址

  - `""` → 空值，即当前导出目录的同 `extension` 文件夹，如

    `{"extension":"epub", "path":""}` 则会在导出文件夹中新建 `epub` 文件夹。

  - `"/xxxx"` → 有值，根据对应值的位置导出。

- type → 导出类型（ 此功能为 [同步助手 1.1.3 版](Sync) 功能，如低于此版本 [请升级](#下载) ）。

  - `annote` → 仅标注 e.g. `1234-xxx@annote.md`

  - `unread` → 仅稍后读  e.g. `1234-xxx.md` **1.3.0 新增**

  - `article` → 仅普通文章  e.g. `xxx.md` **1.3.0 新增**

  - `page` → 全文 e.g. 稍后读 `1234-xxx` 或普通文章 `xxx` （ **legacy 1.3.0 去除** 不建议使用，请使用两个新的参数 `unread`或 `article`  ）

  > type 参数可以省略，当省略后不区分 **普通文章 / 稍后读 / 标注**。

> [!TIP]
> 1.3.0 版增加了新的参数 `unread` `article` 如果你需要区分普通文章（非稍后读）与稍后读以及标注的话，请使用下面的方式。

一个例子

```
{"extension":"assets", "path":"/Users/xxx/Obsidian/Inbox", "type":"article"}
{"extension":"assets", "path":"/Users/xxx/Obsidian/Annote", "type":"annote"}
{"extension":"assets", "path":"/Users/xxx/Obsidian/SimpRead", "type":"unread"}
```

通过上面的设置，可以做到：

- 将普通文章保存到 Inbox 文件夹

- 将稍后读保存到 SimpRead 文件夹

- 将标注保存到 Annote 文件夹

## 注意

如果是 Windows 用户的话，请使用下面的方案（留意有两个斜杠 `\\`）

```
{"extension":"md", "path":"C:\\Users\\***\\Documents\\***\\Desktop\\Documents"}
```

**Reply by Robertxjf** [reply]
我在配置path 的时候失败了，这个是填绝对路径就可以吗?  我看到上面的路径是相对的，那么是相对哪一个的呢
是不是在同步助手中打开增强导出相关功能，然后把上面的配置文件写进去保存，这个功能在导出文章的时候就可以生效了呢

**Reply by Kenshin** [reply]
嗨

1. 需要绝对路径。
2. 没有相对路径，而且这个例子中也没有提到相对路径。😂
3. 同步助手的增强导出，可以看 [此教程](https://kb.simpread.pro/#/page/%E5%A2%9E%E5%BC%BA%E5%AF%BC%E5%87%BA)

**Reply by Robertxjf** [reply]
btw，谢谢回复和你的时间! 我研究了半天搞清楚了原因，也请更新一下文档让后来者注意一下

windows的路径 全是 '\' 而不是像浏览器url和linux一样的 ‘/’ ，直接复制资源管理器路径会出错。 所以所有windows的都得改成 ‘/’

**Reply by Kenshin** [reply]
哈哈，这只是个例子。
Windows 路径至少也是从 `<盘符>:\xxx` 开始的吧。😂

**Reply by YB-Kang** [reply]
我也失败了，我设置自动导出html和markdown，全部都是导出到output文件夹下面，
![image](https://user-images.githubusercontent.com/55050981/178095182-4d6bfff1-099e-4bf0-a2a3-67a550f74f92.png)

**Reply by Kenshin** [reply]
@YB-Kang 

设置导出目录，确保 `path` 对应的目录确定正确且存在。

**Reply by Robertxjf** [reply]
> 我也失败了，我设置自动导出html和markdown，全部都是导出到output文件夹下面， ![image](https://user-images.githubusercontent.com/55050981/178095182-4d6bfff1-099e-4bf0-a2a3-67a550f74f92.png)

和我一样的问题，windows 的path 反斜杠全部要改成正斜杠；这个需要特别注意！

**Reply by YB-Kang** [reply]
是的，我后来成功了

**Reply by Kaergo** [reply]
> > 我也失败了，我设置自动导出html和markdown，全部都是导出到output文件夹下面， ![image](https://user-images.githubusercontent.com/55050981/178095182-4d6bfff1-099e-4bf0-a2a3-67a550f74f92.png)
> 
> 和我一样的问题，windows 的path 反斜杠全部要改成正斜杠；这个需要特别注意！

解决了？请问是用几根斜杠？

**Reply by Kenshin** [reply]
@Kaergo 

文中不是有提示吗？两个 `\\`

**Reply by iblogc** [reply]
`output` → 导出文件夹地址

是不是应该改为

`path`→ 导出文件夹地址

**Reply by Kenshin** [reply]
@iblogc 

谢谢，已修复~

**Reply by laviisa** [reply]
同时导出网页md和标注md，但只需要把标注md导出到指定文件夹，该怎么办？

**Reply by Kenshin** [reply]
@laviisa 

> 同时导出网页md和标注md，但只需要把标注md导出到指定文件夹，该怎么办？

暂时没法实现，不过已经在考虑了。

但你可以使用 Hazel 等自动化工具实现。

**Reply by Kenshin** [reply]
@laviisa

> [同步助手 1.1.3 版](https://github.com/Kenshin/simpread/discussions/4049#discussioncomment-4192563)  发布啦  🎉 
> 
> 内置了此功能，详细说明 [增强导出](https://github.com/Kenshin/simpread/discussions/2958)。

  ```
  {"extension":"assets", "path":"/Users/xxxx/Documents/Obsidian@simpread/Inbox", "type":"page"}
  {"extension":"md", "path":"/Users/xxxx/Documents/Obsidian@simpread/SimpRead", "type":"annote"}
  ```

### 除此之外比较重要的更新

- `assets + md` 对媒体资源（图片）位置的定制化，需要配合以下两个插件使用：

  - 空白 [导入到 Obsidian](https://simpread.ksria.cn/plugins/details/1VQ19jCD8Z)

  - 空白 [导出 md + assets 格式](https://simpread.ksria.cn/plugins/details/BvjrJA6eh5)

- 读取本地文件，可将本地任何 `.html` `.md` 加入到稍后读，详细说明 [请看这里](http://ksria.com/simpread/docs/#/外部文件)。

- 稍后读外部访问，方便第三方工具接入，详细说明 [请看这里](http://ksria.com/simpread/docs/#/稍后读外部访问)。

- 更换文档地址源，方便不同网络环境的用户使用 [请看这里](http://ksria.com/simpread/docs/#/Sync?id=文档地址源)。

- [浅色主题](http://ksria.com/simpread/docs/#/Sync?id=主题) 以及 [自动暗色模式](http://ksria.com/simpread/docs/#/Sync?id=暗色模式)。

- 同步助手的增强导出规则 `type`，可区分全文和标注，[详细说明](http://ksria.com/simpread/docs/#/Sync?id=增强导出)

详细说明 [请看这里](https://github.com/Kenshin/simpread/discussions/4049#discussioncomment-4192563)。

**Comment by Kenshin** [comment]
# 删除旧文件

当 **使用 [知识库](https://www.yuque.com/kenshin/simpread/lglfy2) 时，请启用此功能**。当删除稍后读时，无法自动同步删除本地快照，启用此功能后会避免出现重复的 `idx-xxx.html`。
